operator/src/main/java/org/keycloak/operator/v2alpha1/RealmJob.java

```diff
@@ -0,0 +1,165 @@
+package org.keycloak.operator.v2alpha1;
+
+import io.fabric8.kubernetes.api.model.*;
+import io.fabric8.kubernetes.api.model.apps.Deployment;
+import io.fabric8.kubernetes.api.model.batch.v1.Job;
+import io.fabric8.kubernetes.api.model.batch.v1.JobBuilder;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import org.jboss.logging.Logger;
+import org.keycloak.operator.v2alpha1.crds.realm.RealmStatus;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+import static org.keycloak.operator.Constants.*;
+import static org.keycloak.operator.v2alpha1.NameUtils.getValidDNSSubdomainName;
+
+public class RealmJob {
+
+    private final static Logger logger = Logger.getLogger(RealmJob.class);
+
+    private KubernetesClient client;
+
+    private Consumer<Supplier<Job>> createJobFn = (jb) -> {
+        var job = jb.get();
+        client
+                .batch()
+                .v1()
+                .jobs()
+                .create(job);
+    };
+
+
+    public RealmJob(KubernetesClient client) {
+        this.client = client;
+    }
+
+    private Job buildJob(String name, String namespace, String realmCRName, Container keycloakContainer, Volume secretVolume, List<OwnerReference> ownerReferences) {
+        return new JobBuilder()
+                .withNewMetadata()
+                .withName(name)
+                .withNamespace(namespace)
+                .addToLabels(PART_OF_LABEL, realmCRName)
+                .addToLabels(MANAGED_BY_LABEL, MANAGED_BY_VALUE)
+                .withOwnerReferences(ownerReferences)
+                .endMetadata()
+                .withNewSpec()
+                .withNewTemplate()
+                .withNewSpec()
+                .withContainers(keycloakContainer)
+                .addToVolumes(secretVolume)
+                .withRestartPolicy("Never")
+                .endSpec()
+                .endTemplate()
+                .endSpec()
+                .build();
+    }
+
+    private Volume buildSecretVolume(String volumeName, String secretName) {
+        return new VolumeBuilder()
+                .withName(volumeName)
+                .withSecret(new SecretVolumeSourceBuilder()
+                        .withSecretName(secretName)
+                        .build())
+                .build();
+    }
+
+    private Container buildKeycloakJobContainer(PodTemplateSpec spec, String realmName, String volumeName) {
+        var keycloakContainer = spec.getSpec().getContainers().get(0);
+
+        var importMntPath = "/mnt/realm-import/";
+
+        var dbOptions = keycloakContainer
```

Comment:
this might be different as soon as we have a fix for #9171 @vmuzikar if you have anticipations I can start to integrate those